filebeat/fileset/modules.go

```diff
@@ -51,12 +51,17 @@ func newModuleRegistry(modulesPath string,
 	moduleConfigs []*ModuleConfig,
 	overrides *ModuleOverrides,
 	beatInfo beat.Info,
+	enableAllFilesets bool,
 ) (*ModuleRegistry, error) {
 	reg := ModuleRegistry{
 		registry: []Module{},
 		log:      logp.NewLogger(logName),
 	}
+
 	for _, mcfg := range moduleConfigs {
+		if enableAllFilesets && mcfg.Module != "" && mcfg.Enabled != nil {
```

Comment:
This (and fileset below) looks right to me but reads a little counterintuitive without the context that `mcfg.Enabled == nil` is treated equivalently to `*mcfg.Enabled == true`, so a clarifying comment might be nice. (This could also maybe be simplified to just `if enableAllFilesets { mcfg.Enabled = nil }`? But I'm mixed on whether that's actually clearer or not, the comment is probably the important thing.)

Alternatively if we could avoid mutating the config entirely by skipping this `if` and changing the second one to `if mcfg.Module == "" || (mcfg.Enabled != nil && !(mcfg.Enabled) && !enableAllFilesets)` that would be nice, but I'm not confident off the top of my head that the `Enabled` field doesn't propagate anywhere else so feel free to ignore that option if you also aren't sure ^^

Reply:
added comment.

Yeah, this is odd.  The real problem is that a ModuleConfig (and Fileset) with all default values can reach this code.  So you have to check to make sure the Module name is not the default value "", and that the Enabled pointer isn't nil.  If those are "real" then you probably have a config from an actual module.